sound/soc/sof/sof-client.c

```diff
@@ -35,6 +36,20 @@ static int sof_client_dev_add_data(struct sof_client_dev *cdev, const void *data
 	return 0;
 }
 
+int sof_register_clients(struct snd_sof_dev *sdev)
+{
+	if (sof_ops(sdev) && sof_ops(sdev)->ipc_register_clients)
+		return sof_ops(sdev)->ipc_register_clients(sdev);
```

Comment:
Yes, that's true, but for the next patch to convert the IPC flood test to a client I need will need a placeholder where the non platform dependent clients can be registered.
We will eventually going to need this and it is just adds the needed infra for the probes conversion which is going to happen after some churn.

I think it would look much worst if I split this patch up and squash half of it to the next patch and then add the callback support on top of it.

If you insist, I will of course do it.